mitmproxy/addons/replace.py

```diff
@@ -38,7 +38,9 @@ def parse_hook(s):
         patt, a, b = parts
     else:
         raise exceptions.OptionsError(
-            "Invalid replacement specifier: %s" % s
+            "Invalid replacement specifier: {}\n"
+            "If you want to use mitmproxy in reverse proxy mode. "
+            "Use --mode reverse:SPEC instead".format(s)
```

Comment:
This seems to be the wrong place to me. I think we should rather check for "-R" followed by something that starts with "http"  in main.py so that we don't spread out those legacy checks over the whole codebase.